--- CMakeLists.txt	2023-02-22 17:27:23.619115753 -0600
+++ CMakeLists.txt.orig	2023-02-22 17:26:46.480451170 -0600
@@ -16,7 +16,6 @@
 include(GNUInstallDirs)
 
 find_package(PkgConfig REQUIRED)
-find_package(pybind11 CONFIG)
 
 include(GNUInstallDirs)
 include(FindBZip2)
@@ -455,12 +454,6 @@
 
 endif()
 
-# ---- Python bindings - mostly for testing at this stage
-pybind11_add_module(pyvcflib "${CMAKE_SOURCE_DIR}/src/pythonffi.cpp")
-add_dependencies(pyvcflib ${vcflib_DEPS})
-target_link_libraries(pyvcflib PUBLIC vcflib ${vcflib_LIBS} ${WFALIB})
-install(TARGETS pyvcflib LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-
 # ---- Test
 
 enable_testing()
